### ISSUE TITLE
DaoClassesTest fails on Java 11 with "java.lang.IllegalArgumentException: object is not an instance of declaring class"

### ISSUE THREAD

**Issue by odockal** [redhat-developer/rsp-server#274]
```
java.lang.IllegalArgumentException: object is not an instance of declaring class
	at org.jboss.tools.rsp.schema.DaoClassesTest.addJarToClasspath(DaoClassesTest.java:106)
	at org.jboss.tools.rsp.schema.DaoClassesTest.shouldDiscoverAllInPackageInJar(DaoClassesTest.java:69)
```

**Comment by odockal** [comment]
When I follow the exception and change 
```
```
to
```
```
I got 
```

**Comment by adietish** [comment]
According to https://blog.codefx.org/java/java-9-migration-guide/ "Casting To URL Class Loader" the AppClassloader in jdk11 is no URLClassloader any more. So we need to find a different way to have the jar - that we require for testing - in the classpath. 
Prior to this I used the following in the pom, which I dropped because this caused warnings in the maven build:
```
     <dependency>
         <groupId>test</groupId>
         <artifactId>test</artifactId>
         <scope>system</scope>
         <version>1.0</version>
         <systemPath>${basedir}/src/test/resources/daoclasses.jar</systemPath>
      </dependency>
```

**Comment by adietish** [comment]
fixed thanks to #275. Closing.
